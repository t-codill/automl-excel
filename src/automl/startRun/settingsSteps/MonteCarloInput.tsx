import * as React from "react";
import { CrossValidationNumberInput } from "./CrossValidationNumberInput";
import { PercentageValidationInput } from "./PercentageValidationInput";

export class MonteCarloInput extends React.Component {
    public render(): React.ReactNode {
        return <>
            <CrossValidationNumberInput />
            <PercentageValidationInput />
        </>;
    }
}
