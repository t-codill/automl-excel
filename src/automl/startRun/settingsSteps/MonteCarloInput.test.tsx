import { shallow } from "enzyme";
import * as React from "react";
import { MonteCarloInput } from "./MonteCarloInput";

describe("Cross Validation Input", () => {
    it("should render with validation error undefined props", () => {
        const c = {
            percentageValidation: 0,
            crossValidationNumber: 0
        };
        const tree = shallow(
            <MonteCarloInput {...c} />
        );
        expect(tree)
            .toMatchSnapshot();
    });

    it("should render with valid props", () => {
        const c = {
            percentageValidation: 5,
            crossValidationNumber: 20
        };
        const tree = shallow(
            <MonteCarloInput {...c} />
        );
        expect(tree)
            .toMatchSnapshot();
    });
});
