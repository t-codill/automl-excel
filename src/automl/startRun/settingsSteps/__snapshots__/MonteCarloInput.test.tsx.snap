// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cross Validation Input should render with valid props 1`] = `
<Fragment>
  <crossValidationNumberInput />
  <percentageValidationInput />
</Fragment>
`;

exports[`Cross Validation Input should render with validation error undefined props 1`] = `
<Fragment>
  <crossValidationNumberInput />
  <percentageValidationInput />
</Fragment>
`;
